src/js/utils/spec-helpers/ajax-helpers.js

```diff
@@ -17,204 +28,264 @@ define(['sinon', 'underscore', 'URI'], function(sinon, _, URI) {
+     * @param {string} method the expected method of the request
+     * @param {string} url the expected url of the request
+     * @param {object} jsonRequest the expected request body as an object
+     */
+    expectJsonRequest = function(requests, method, url, jsonRequest) {
+        var request = currentRequest(requests);
+        expect(request.readyState).toEqual(XML_HTTP_READY_STATES.OPENED);
         expect(request.url).toEqual(url);
         expect(request.method).toEqual(method);
-        expect(JSON.parse(request.requestBody)).toEqual(jsonBody);
+        expect(JSON.parse(request.requestBody)).toEqual(jsonRequest);
     };
 
     /**
      * Expect that a JSON request be made with the given URL and parameters.
      *
-     * @param requests The collected requests
-     * @param expectedUrl The expected URL excluding the parameters
-     * @param expectedParameters An object representing the URL parameters
-     * @param requestIndex An optional index for the request (by default, the last request is used)
+     * @param {object} requests The collected requests
+     * @param {string} expectedUrl The expected URL excluding the parameters
+     * @param {object} expectedParameters An object representing the URL parameters
      */
-    expectRequestURL = function(requests, expectedUrl, expectedParameters, requestIndex) {
-        var request, parameters;
-        if (_.isUndefined(requestIndex)) {
-            requestIndex = requests.length - 1;
-        }
-        request = requests[requestIndex];
+    expectRequestURL = function(requests, expectedUrl, expectedParameters) {
+        var request = currentRequest(requests),
+            parameters;
         expect(new URI(request.url).path()).toEqual(expectedUrl);
         parameters = new URI(request.url).query(true);
         delete parameters._;  // Ignore the cache-busting argument
         expect(parameters).toEqual(expectedParameters);
     };
 
     /**
-     * Intended for use with POST requests using
-     * application/x-www-form-urlencoded.
+     * Intended for use with POST requests using application/x-www-form-urlencoded.
      *
-     * @param requests an array of fired sinon requests
-     * @param url the expected url of the request
-     * @param body the expected body of the request
-     * @param requestIndex the expected index of the request in the
-     *     array (higher indices occur later).  Defaults to
-     *     requests.length - 1.
+     * @param {object} requests an array of fired sinon requests
+     * @param {string} url the expected url of the request
+     * @param {string} body the expected body of the request
      */
-    expectPostRequest = function(requests, url, body, requestIndex) {
-        var request;
-        if (_.isUndefined(requestIndex)) {
-            requestIndex = requests.length - 1;
-        }
-        request = requests[requestIndex];
+    expectPostRequest = function(requests, url, body) {
```

Comment:
We should implement this on top of `expectRequest` to avoid duplicated logic:

```javascript
expectPostRequest = function(requests, url, body) {
    return expectRequest(requests, 'POST', url, body);
};
```